crypto/x509/x509_lu.c

```diff
@@ -141,10 +144,12 @@ static int x509_object_cmp(const X509_OBJECT *const *a,
 
 X509_STORE *X509_STORE_new(void)
 {
-    X509_STORE *ret;
+    X509_STORE *ret = OPENSSL_zalloc(sizeof(*ret));
 
-    if ((ret = OPENSSL_zalloc(sizeof(*ret))) == NULL)
+    if (ret == NULL) {
+        X509err(X509_F_X509_STORE_NEW, ERR_R_MALLOC_FAILURE);
         return NULL;
+    }
     if ((ret->objs = sk_X509_OBJECT_new(x509_object_cmp)) == NULL)
         goto err;
```

Comment:
The  `X509err(X509_F_X509_STORE_NEW, ERR_R_MALLOC_FAILURE)` should go here, before the goto...